utils/utils.go

```diff
@@ -39,10 +40,14 @@ func DownloadWithProgress(url string, tarName string, destFolder string) (err er
 		resp.ContentLength,
 		"Downloading",
 	)
-	_, err = io.Copy(io.MultiWriter(f, bar), resp.Body)
+	bytes, err := io.Copy(io.MultiWriter(f, bar), resp.Body)
 	if err != nil {
 		return err
 	}
+	// a subjective limit on bytes should be greater than 10000 (0.01 mb) to be considered a file that we expect
+	if bytes < 10000 {
```

Comment:
To be honest, it is not a clever solution because it breaks every single principle of engineering.

Function `DownloadWithProgress` should be renamed `DownloadGoTarWithProgress` to at least maintain single responsibility principle.


Please don't hate!
If you are good, then please merge it

Reply:
@kevincobain2000 why you do not check `resp.StatusCode`?

If we get file NotFound we have 404 StatusCode. And we have 200 StatusCode if file exist.

For example on line 29 or 33.

I think it more correct.

Reply:
That too. I ll check if the response code is indeed 404 and not 301 to an error page.